xonsh/main.py

```diff
@@ -252,10 +253,58 @@ def premain(argv=None):
     return args
 
 
+def _failback_to_other_shells(argv):
+    args = None
+    try:
+        args = premain(argv)
+    except Exception:
+        pass
+    # only failback for interactive shell; if we cannot tell, treat it
+    # as an interactive one for safe.
+    if hasattr(args, 'mode') and args.mode != XonshMode.interactive:
+        sys.exit(1)
+        # put a harmless verbose `return` here so that unit test is able to
+        # break here when mock up `sys.exit()`
+        return
+
+    print('Xonsh encountered an issue during launch', file=sys.stderr)
+    foreign_shell = None
+    shells_file = '/etc/shells'
+    if not os.path.exists(shells_file):
+        # right now, it will always return here on Windows
+        return
+    excluded_list = ['xonsh', 'screen']
+    with open(shells_file) as f:
+        for line in f:
+            line = line.strip()
+            if not line or line.startswith('#'):
+                continue
+            if '/' not in line:
+                continue
+            _, shell = line.rsplit('/', 1)
+            if shell in excluded_list:
+                continue
+            if not os.path.exists(line):
+                continue
+            foreign_shell = line
+            break
+    if foreign_shell:
+        print('Failback to {}'.format(foreign_shell), file=sys.stderr)
+        os.execlp(foreign_shell, foreign_shell)
```

Comment:
same reason here I guess. what errors should printed should have already printed.